quasar-core/src/main/java/co/paralleluniverse/strands/channels/disruptor/StrandBlockingWaitStrategy.java

```diff
@@ -57,11 +53,13 @@ public long waitFor1(long sequence, Sequence cursorSequence, Sequence dependentS
             try {
                 for (int i = 0; (availableSequence = cursorSequence.get()) < sequence; i++) {
                     barrier.checkAlert();
-                    processorNotifyCondition.await(i);
+                    processorNotifyCondition.await(i, 1, TimeUnit.NANOSECONDS);
```

Comment:
But when fibers block they don't block the thread they run in (i.e. the FJPool does execute other tasks). Have you tried using Quasar with plain channels? They work the same way. If you haven't I suggest you do, just to make sure everything works as it should.